File: src/CCC/c19/S1.py
flips = input()

hCount = 0
vCount = 0

for c in flips:
    if c == "H":
        hCount += 1
    else:
        vCount += 1

hCount %= 2
vCount %= 2
if hCount == 0 and vCount == 0:
    print("1 2")
    print("3 4")
elif hCount == 0 and vCount == 1:
    print("2 1")
    print("4 3")
elif hCount == 1 and vCount == 0:
    print("3 4")
    print("1 2")
else:
    print("4 3")
    print("2 1")

File: src/CCC/c19/S1.cpp
#include <iostream>
#include <string>

using namespace std;
int main(void){
    string line;
    getline(cin,line);

    int hCount=0;
    int vCount=0;

    for (char c:line){
        if (c=='H'){
            hCount++;
        }else{
            vCount++;
        }
    }

    hCount%=2;
    vCount%=2;

    if (hCount == 0 && vCount == 0) {
        cout<<"1 2"<<endl;
        cout<<"3 4"<<endl;
    } else if (hCount == 0 && vCount == 1) {
        cout<<"2 1"<<endl;
        cout<<"4 3"<<endl;
    } else if (hCount == 1 && vCount == 0) {
        cout<<"3 4"<<endl;
        cout<<"1 2"<<endl;
    } else {
        cout<<"4 3"<<endl;
        cout<<"2 1"<<endl;
    }

    return 0;
}
